### ISSUE TITLE
How to calculate MPJPE based on your code?

### ISSUE THREAD

**Issue by RichardChen20** [bastianwandt/CanonPose#3]
I'm very interested in your work, I tried to train the network using GT 2D poses following your paper and codes, but the MPJPE seems to be much larger (90mm) than that (65mm) reported in your paper. I noticed you have done the data normalization to the input poses and the scale of output poses is small, may I ask how you calculate the MPJPE? Can you provide the code?

**Comment by bastianwandt** [comment]
Hi, you need detections from Alphapose. These go through the skeleton morphing network and then to the model you trained. The output is not scaled correctly and therefore we used the scaled MPJPE as described in the paper. That means a predicted pose needs to be scaled first before calculating the MPJPE. The MPJPE is calculated as the mean euclidean distance between joints.
I hope that helps.

**Comment by RichardChen20** [comment]
Could you please tell me more about your scaled MPJPE. Here are my steps, I first do the centering and F norm to the input GT 2D poses and target 3D poses and I store the F norm of 3D poses. When I get the predicted 3D poses, I do this:
MPJPE = (((rot_poses - target) * scale.unsqueeze(1)) ** 2).reshape(-1, 3, 17).sum(axis=1).sqrt().mean()
Is this way different from your scaled MPJPE?

**Comment by bastianwandt** [comment]
You need to scale rot_poses first and then calculate the distance. Also your skeleton seems to be different.

**Comment by RichardChen20** [comment]
Do you mean that I need to scale rot_poses and target to the same range like (-1,1)  and then scale the error to original size of initial target, then correct MPJPE can be calculated? I would appreciate it so much if you could reply!

**Comment by sirfoga** [comment]
What you need to calculate is called [NMPJPE](https://paperswithcode.com/paper/neural-scene-decomposition-for-multi-person/review/) (i.e MPJPE scaled wrt to ground-truth data's scale): you find the scaling factor (scalar constant) between your prediction and ground-truth data and scale by that factor. See [my implementation](https://github.com/sirfoga/learnable-triangulation-pytorch/blob/master/mvn/utils/tred.py#L282) based on [Umeyama approach](https://ieeexplore.ieee.org/document/88573)

**Comment by bastianwandt** [comment]
Thanks @sirfoga. That's exactly what I'm doing. It's the same scaling step as the one during Procrustes alignment.

**Comment by RichardChen20** [comment]
Thank you all very much! It seems that I can close this issue.
